src/main/java/engine/Calculator.java

```diff
@@ -0,0 +1,112 @@
+package engine;
+
+import engine.io.*;
+import engine.repository.CalculatorRepository;
+
+import java.util.List;
+import java.util.Stack;
+
+public class Calculator implements Runnable {
+
+    private final CalculatorRepository repository;
+
+    private final Input input;
+    private final Output output;
+
+    public Calculator(CalculatorRepository repository, Input input, Output output) {
+        this.repository = repository;
+        this.input = input;
+        this.output = output;
+    }
+
+    @Override
+    public void run() {
+
+        while (true) {
+            String inputString = input.initialInput();
+
+            int inputValue = Integer.parseInt(inputString);
+
+            try {
+                switch (inputValue) {
+                    case 1:     //조회
+                        List<String> formulaList = repository.findAllValues();
+                        for (String f : formulaList) {
+                            output.formula(f);
+                        }
+                        break;
+                    case 2:     //계산
+                        String inputFormula = input.input();
+                        String answer = cal(inputFormula);
+                        output.output(answer);
+                        save(inputFormula, answer);
+                        break;
+                    default:
+                        output.inputError();
+                }
+            } catch (ArithmeticException e) {
+                output.inputError();
+            }
+        }
+
+    }
+
+    public String cal(String input) {
+        String formula[] = parse(input);
+
+        /** 후위 계산식으로 변경 **/
+        Stack<Double> operandStack = new Stack<>();
+        Stack<String> operatorStack = new Stack<>();
+
+        double answer = Double.valueOf(formula[0]);
+        operandStack.push(answer);
+
+        for (int i = 1; i < formula.length; i += 2) {
+            String pushOper = formula[i];
+            double operand = Double.valueOf(formula[i + 1]);
```

Comment:
```suggestion
            double operand = Double.parseDouble(formula[i + 1]);
```

valueOf와 parseDouble은 차이점이 조금 있는데요.
내용 한번 확인해보시면 좋을 듯 합니다.


[Double.valueOf(s) vs. Double.parseDouble [duplicate]](https://stackoverflow.com/questions/7255078/double-valueofs-vs-double-parsedouble)